### PR TITLE
Now gives you comments when given a review_id, Completing pt.6

### PR DESCRIPTION
**Pull request by ERose18**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -126,4 +126,61 @@ describe('GET /api/reviews', () => {
...
 
- 
\ No newline at end of file
+ describe('GET /api/reviews/:review_id/comments', () => {
+    test('200: Ensures that what is passed IS an Object', () =>{ 
+        return request(app)
+        .get('/api/reviews/3/comments')
+        .expect(200)
+        .then(({body}) => {
+            console.log(body)
```
Comment (callumu21) [inline __tests__/app.test.js]: Just make sure to remove any stray console.logs when you're ready to push up

File: __tests__/app.test.js [hunk 2]
```diff
@@ -126,4 +126,61 @@ describe('GET /api/reviews', () => {
...
+            const comments = body.comments;
+            expect(comments).toBeInstanceOf(Object);
+     });
+    });
+    test('200: Returns the length of the Object passed', () =>{
```
Comment (callumu21) [inline __tests__/app.test.js]: I see what you're doing here - building up from an array, to an array with a set length, to an array of objects. However, your test on line 158 implicitly checks that it's an array with the .forEach method, so we can remove these previous ones once we have confidence that it's covered by later tests.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -126,4 +126,61 @@ describe('GET /api/reviews', () => {
...
+            expect(comments).toEqual([]);
+     });
+    });
+    test('200: Returns an array of comments from the given review_id', () =>{
+        return request(app)
+        .get('/api/reviews/3/comments')
+        .expect(200)
+        .then(({body}) => {
+            const comments = body.comments;
+            console.log(comments)
```
Comment (callumu21) [inline __tests__/app.test.js]: We can also remove this console.log

File: __tests__/app.test.js [hunk 4]
```diff
@@ -126,4 +126,61 @@ describe('GET /api/reviews', () => {
...
+            comments.forEach((comment) => {
+                expect(comment).toMatchObject({
+                    comment_id: expect.any(Number),
+                    votes: expect.any(Number),
+                    created_at: expect.any(String),
+                    author: expect.any(String),
+                    body: expect.any(String),
+                    review_id: 3,
```
Comment (callumu21) [inline __tests__/app.test.js:153]: Nice - remembering to check against the specific review_id

File: __tests__/app.test.js [hunk 5]
```diff
@@ -126,4 +126,61 @@ describe('GET /api/reviews', () => {
...
+    test('200: Returns the length of the Object passed', () =>{ 
+        return request(app)
+        .get('/api/reviews/3/comments')
+        .expect(200)
+        .then(({body}) => {
+            const comments = body.comments;
+            expect(comments.length).toBe(3);
+     });
+    });
+    test('200: Returns an empty array of comments when passed a review_id with no comments', () =>{
```
Comment (callumu21) [inline __tests__/app.test.js:130]: Really nice! This is one that people often forget!

File: __tests__/app.test.js [hunk 6]
```diff
@@ -126,4 +126,61 @@ describe('GET /api/reviews', () => {
...
+         });
+    });  
+    test('404: Returns "ID Not Found" if given an "ID" that isnt in the database', () =>{
+        return request(app)
+        .get('/api/reviews/9000/comments')
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe('ID Not Found');
+        });
+    });
```
Comment (callumu21) [inline __tests__/app.test.js:166]: Just missing a test for when a client attempts to make a request for the comments of a review with an invalid ID ('pineapple') 👍

File: controllers/get-reviews-comments-controllers.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+const reviews = require('../db/data/test-data/reviews');
+  const comments = require('../db/data/test-data/comments');
+  const {fetchReviewsByID} = require('../models/get-reviews-id-models')
+  const {fetchReviewCommentsByID} = require('../models/get-reviews-comments-models');
+  
+  function getReviewCommentsByID(request, response, next){
+      const {review_id} = request.params;
+      const promiseForComment = [fetchReviewCommentsByID(review_id)];
+  
+      if(review_id){promiseForComment.push(fetchReviewsByID(review_id))};
```
Comment (callumu21) [inline controllers/get-reviews-comments-controllers.js:10]: I understand this if statement, but you're also going to run fetchReviewCommentsByID if there isn't a valid review_id, so not sure it's completely needed here.

**Review by callumu21 — CHANGES_REQUESTED** [review]
Really nice start! Just missing a test for an invalid review_id.

I've also left some comments on small refactors and stray console.logs, but let me know if anything isn't clear.

Once you've added everything in, please resubmit this PR 👍

File: __tests__/app.test.js (hunk 6 above)
Comment (HeatherEver) [reply __tests__/app.test.js:166]: 🍍

**Review by HeatherEver — APPROVED** [review]
Nice changes :) merge away

**Pull request merged**